ek9000App/src/scripts/generate.py

```diff
@@ -183,35 +196,110 @@ def write(self, fp):
                 fp.write(f'\tfield({i[0]}, "{i[1]}")\n')
             fp.write('}\n\n')
 
+
+    @staticmethod
+    def _get_raw_range(spec: SpecType) -> Tuple[int,int]:
+        """
+        Computes the raw range for the terminal
+        
+        Returns
+        -------
+        Tuple[int,int]:
+            Range interval (min,max)
+        """
+        isSigned = False
+        match spec['representation']:
+            case 'int16':
+                isSigned = True
+            case 'uint16':
+                isSigned = False
+            case other:
+                assert False
+        max = 32767
+        min = -max if spec['min'] < 0 and isSigned else 0 # Unsigned terminals still have range 0-32767
+        return (min, max)
+
+
+    @staticmethod
+    def _compute_deadbands(spec: SpecType) -> float:
+        """
+        Compute deadband parameters ADEL and MDEL based on max EGU and error specified in datasheet
+        Most terminals are rated for measurement error of <=0.3%
+        
+        Returns
+        -------
+        float:
+            The value ADEL and MDEL should be set to
+        """
+        error = spec['error'] / 100.0 # error is in %, convert that to float
+        return spec['max'] * error # Docs specify that error is relative to the full scale value
+
+
+    def _compute_analog_defaults(self) -> dict:
+        """
+        Compute ESLO, EOFF, EGUF and EGUL for the analog I/O terminal
+        
+        Returns
+        -------
+        dict:
+            Mapping of PV -> values
+        """
+
+        if self.spec is None:
+            return {}
+
+        rawMin, rawMax = self._get_raw_range(self.spec)
+        range = self.spec['max'] - self.spec['min']
+        deadband = self._compute_deadbands(self.spec)
+        return {
+            'ESLO': (range / (rawMax-rawMin)),
+            'EOFF': (rawMax * self.spec['min'] - rawMin * self.spec['max']) / (rawMax-rawMin),
+            'EGUF': self.spec['max'],
+            'EGUL': self.spec['min'],
+            'EGU': self.spec['egu'],
+            'LINR': 'LINEAR',
```

Comment:
Yep, that's correct. I should've set it to be `SLOPE`. Interestingly it doesn't seem to matter if it's set to `LINR` or `SLOPE`, I think because the `linconv` routine is a no-op in the device support code.